### ISSUE TITLE
[question] Xcode Not showing header suggestions 

### ISSUE THREAD

**Issue by jaibeer72** [conan-io/conan#14651]
### Xcode not showing header suggestions in project 

Hey! I believe I might be doing something wrong. 

[DESCLAMER: THIS IS MY FIRST TIME USING CONAN, AND I ALSO AM A JUNIOR DEV] 

## Trying to build a basic rendering engine that can build both on Visual Studio and Xcode, and later I want to expand it to other editors if necessary. 

for some reason, I don't get header suggestions for Xcode but on visual studio it's fine. 

my Conan file is done like this 

```python
from conan import ConanFile
from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout

class VulkanGFXConan(ConanFile):
    name = "Cross-Platform-VulkanGFX"
    version = "0.0.1b"
    settings = "os", "compiler", "build_type", "arch"
    requires = [
        "vulkan-loader/1.3.239.0",
        "glm/cci.20230113",
        "glfw/3.3.8"
    ]
    generators = "CMakeDeps", "CMakeToolchain" 

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def package_info(self):
        self.cpp_info.includedirs = ["include", "other_include"]

    def build_requirements(self):
        self.tool_requires("cmake/3.22.6") 
  ```
        
  and my CMakeLists.txt is 
  ```
        # Set Minimum requirements for the project 
cmake_minimum_required(VERSION 3.5)

cmake_policy(SET CMP0091 NEW)
cmake_policy(SET CMP0048 NEW)

# Set the project name
project(Cross-Platform-VulkanGFX VERSION 0.1.0)

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_CXX_EXTENSIONS OFF)

# Set the include directories via conan dependencies
include(${CMAKE_BINARY_DIR}/conan_toolchain.cmake)

# Recursively find all source files
file(GLOB_RECURSE SOURCE_FILES ${CMAKE_SOURCE_DIR}/Source/*.cpp ${CMAKE_SOURCE_DIR}/Source/*.h ${CMAKE_SOURCE_DIR}/Source/*.hpp)

# Use the natural file system in Xcode
source_group(TREE "${CMAKE_SOURCE_DIR}/Source" PREFIX "Source" FILES ${SOURCE_FILES})

# Include dependencies from Conan
list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
find_package(Vulkan REQUIRED)
find_package(glm REQUIRED)
find_package(glfw3 REQUIRED)

set(CMAKE_SKIP_RPATH TRUE)

# Add executable
add_executable(Cross-Platform-VulkanGFX ${SOURCE_FILES})

# Set C++ standard
# set_property(TARGET Cross-Platform-VulkanGFX PROPERTY CXX_STANDARD 11)

# set Cross-Platform-VulkanGFX as the startup project
set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT Cross-Platform-VulkanGFX)

# Link Conan libraries
target_link_libraries(${PROJECT_NAME} PUBLIC Vulkan::Vulkan glm::glm glfw)
target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_SOURCE_DIR}/Source)
``` 

for some reason, my XCode headers don't show up in the suggestions but work well on visual studio on Windows. I have no idea what I am doing wrong. 

This is my repo and the commands I am running are here 
https://github.com/jaibeer72/VulkanGFX/blob/main/MakeProject.sh 

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @jaibeer72 

Thanks for your question! Don't worry about being a newbie, don't hesitate to ask any question you might have :)

> conan install . -s build_type=Debug

Just to be sure, are you sure the selected configuration in Xcode is also ``Debug``? If not you might want to install the ``Release`` dependencies. Did the project build succesfully? Both from the IDE and with ``cmake --build . --config Debug`` in the command line? 

If the project builds successfully, it might be some issue or limitation in Xcode (I don't know about Xcode, I use mostly Windows and Linux).

**Comment by jaibeer72** [comment]
Yeah! so the project builds successfully on both the IDE and, I believe, even on each platform. 
I did run the command `cmake --build . --config Debug` locally, the build seems to be successful (I have not yet got a full renderer or even the full hello world done) I also have a GitHub action that tries to build the exe every time I push on main.

but yeah when I do run the exe there seems to be an issue where I get this 

```
 /Users/jaibeerdugal/Documents/PersonalProject/VulkanGFX/ProjectFolder/Debug/Cross-Platform-VulkanGFX ; exit;
dyld[88842]: Library not loaded: @rpath/libvulkan.1.dylib
  Referenced from: <6FD44C8B-79D5-3D96-A258-831D12AE8ADE> /Users/jaibeerdugal/Documents/PersonalProject/VulkanGFX/ProjectFolder/Debug/Cross-Platform-VulkanGFX
  Reason: tried: '/usr/local/lib/libvulkan.1.dylib' (no such file), '/usr/lib/libvulkan.1.dylib' (no such file, not in dyld cache)
zsh: abort    
  
 ```

https://github.com/jaibeer72/VulkanGFX/actions 
but that's with msBuild and xcodeBuild for all targets. so maybe it does not seem to have the Vulkan libs there

**Comment by memsharded** [comment]
The runtime error is kind of expected, if you are linking dynamically (I think that is the only option with vulkan?), and not activating somehow the ``conanrun.sh`` environment. Conan when ``conan install`` generates 2 files, one for build time, and the other for runtime. This one for runtime should have the DYLD_LIBRARY_PATH env-var defined to point to the folder containing the .dylib for vulkan. Can you please check the file, source it, and try running again?

**Comment by jaibeer72** [comment]
so ! 

the conanrun.sh points to a file that does seem to have 

```
export LD_LIBRARY_PATH="/Users/user/.conan2/p/b/vulkaf54d2f62cea45/p/lib:$LD_LIBRARY_PATH"
export DYLD_LIBRARY_PATH="/Users/user/.conan2/p/b/vulkaf54d2f62cea45/p/lib:$DYLD_LIBRARY_PATH"
```

But it does seem to fail 

<img width="1275" alt="Screenshot 2023-09-04 at 10 01 51" src="https://github.com/conan-io/conan/assets/22143556/bdd48252-f959-4303-860b-3e5f003bf1fd">

I don't remember which was the earlier version where things were a bit better. 
I don't know if the way my confile is set up or my cmake is a mess. 

this is the branch I am working on https://github.com/jaibeer72/VulkanGFX/tree/jaibeer72/jai-12-vulkan-module-management 

I am not sure what I am doing wrong maybe I need to learn more about CMake or something cause all I am trying to do is 
have a CMake with conan to manage my packages so I don't have to have the packages in my git I can just consume them and later make files that I will need in different packages and also do some devops to do some testing .

**Comment by memsharded** [comment]
Quick question: how did you activated (``source conanrun.sh``) the run-environment, and then executed the binary in the IDE? The IDE project typically won't do this automatically. If you didn't, could you please try from the terminal, first activate the ``conanrun.sh``, then run the executable?

**Comment by jaibeer72** [comment]
so ! in my MakeProject.sh I run the source conanrun.sh as well. But it still seems to fail. 
I have tried doing a release build as well. But it's not going that well so basically for some reason Xcode integration with my CMake does not seem to be working at all. 

so things I have tried 
- My MakeProject.sh 
- Running the source conanrun.sh terminal then in the build. 

I did run the project on my latest main on my windows machine as well there it seems to be fine for sure. 
I have genuinely no idea what I am doing wrong at all . 

this is very irksome cause my main on the go device right now is my mac and I would love to be able to work on this anywhere .

**Comment by jaibeer72** [comment]
@memsharded, do you think there would be an example of the conan.py file for MoltenVK available? something I can use to maybe see what I am doing wrong ? 

like a project maybe I can use to guide me through these issues ?

**Comment by jaibeer72** [comment]
all right so the headers are still not showing up but! for some reason, now we have the dynamic libs working.. for some reason simplifying and changing the name of the exe and the project helped.. and I am more confused than ever but till it works I know I should not complaint

anyway had to shift to MoltenVk at some point
